lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -962,47 +962,34 @@ class MongoClientInterface {
 
     internalListObject(bucketName, params, extension, log, cb) {
         const c = this.getCollection(bucketName);
+        const stream = new MongoReadStream(c, params, params.mongifiedSearch);
+        const skip = new Skip({
+            extension,
+            gte: params.gte,
+        });
         let cbDone = false;
-        let streakLength = 0;
-        const stream = new MongoReadStream(c, params,
-                                           params.mongifiedSearch);
+
+        skip.setListingEndCb(() => {
```

Comment:
minor: if those are mandatory callbacks, maybe they should be set directly in the constructor params instead of setters?